Library.java

```diff
@@ -0,0 +1,249 @@
+			    
+					library = (Library) libraryState.readObject();
+					Calendar.getInstance().setDate(library.loanDate);
+					libraryState.close();
+				}
+				catch (Exception e) {
+					throw new RuntimeException(e);
+				}
+			}
+			else library = new Library();
+		}
+		return library;
+	}
+
+	
+	public static synchronized void saveState() {
+		if (library != null) {
+			library.loanDate = Calendar.getInstance().getDate();
+			try (ObjectOutputStream libraryState = new ObjectOutputStream(new FileOutputStream(LIBRARY_FILE));) {
+				libraryState.writeObject(library);
+				libraryState.flush();
+				libraryState.close();	
+			}
+			catch (Exception e) {
+				throw new RuntimeException(e);
+			}
+		}
+	}
+
+	
+	public int getBookID() {
+		return bookID;
+	}
+	
+	
+	public int getMemberID() {
+		return memberID;
+	}
+	
+	
+	private int getNextBookID() {
+		return bookID++;
+	}
+
+	
+	private int getNextMemberID() {
+		return memberID++;
+	}
+
+	
+	private int getNextLoanID() {
+		return loanID++;
+	}
+
+	
+	public List<Member> getMembers() {		
+		return new ArrayList<Member>(members.values()); 
+	}
+
+
+	public List<Book> getBooks() {		
+		return new ArrayList<Book>(catalogue.values()); 
+	}
+
+
+	public List<loan> getCurrentLoans() {
+		return new ArrayList<loan>(currentLoans.values());
+	}
+
+
+	public Member addMember(String lastName, String firstName, String email, int phoneNo) {		
+		Member member = new Member(lastName, firstName, email, phoneNo, getNextMemberID());
+		members.put(member.getID(), member);		
+		return member;
+	}
+
+	
+	public Book addBook(String author, String title, String callNumber) {		
+		Book book = new Book(author, title, callNumber, getNextBookID());
+		catalogue.put(book.getID(), book);		
+		return book;
+	}
+
+	
+	public Member getMember(int memberId) {
+		if (members.containsKey(memberId)){ 
+            return members.get(memberId);
+        }    
+		return null;
+	}
+
+	
+	public Book getBook(int bookId) {
+		if (catalogue.containsKey(bookId)) {
+            return catalogue.get(bookId);
+        }		
+		return null;
+	}
+
+	
+	public int getLoanLimit() {
+		return LOAN_LIMIT;
+	}
+
+	
+	public boolean approveLoan(Member member) {		
+        if (member.numberOfCurrentLoans() == LOAN_LIMIT ) {
+			return false;
+        }
+		if (member.finesOwed() >= MAX_FINES_OWED) {
+			return false;
+		}
+		for (loan loan : member.getLoans()) {
+			if (loan.OVer_Due()) {
+				return false;
+            }
+        }
+		return true;
+	}
+
+	
+	public int getRemainingLoans(Member member) {		
+		return LOAN_LIMIT - member.numberOfCurrentLoans();
+	}
+
+	
+	public loan issueLoan(Book book, Member member) {
+		Date dueDate = Calendar.getInstance().dueDate(LOAN_PERIOD);
+		loan loan = new loan(getNextLoanID(), book, member, dueDate);
+		member.takeOutLoan(loan);
+		book.borrowBook();
+		loans.put(loan.ID(), loan);
+		currentLoans.put(book.getID(), loan);

```

Comment:
Please separate method within method call e.g.

```suggestion
		int bookID = book.getID()
		currentLoans.put(bookID, loan);
```